api/tableau_api/views.py

```diff
@@ -8,9 +37,122 @@ class SummaryStatisticDailyStudyView(TableauApiView):
+            order_direction (str): order to sort in, either "ascending" or "descending"
+            participant_ids (optional[list[str]]): a list of participants to limit the search to
+
+        Returns (queryset[SummaryStatisticsDaily]): the SummaryStatisticsDaily objects specified by the parameters
+        """
+        if order_direction == 'descending':
+            ordered_by = '-' + ordered_by
+        queryset = SummaryStatisticDaily.objects.filter(study__object_id=study_id).filter(deleted=False)
+        if participant_ids:
+            queryset = queryset.filter(participant__patient_id__in=participant_ids)
+        if end_date:
+            queryset = queryset.filter(date__lte=end_date)
+        if start_date:
+            queryset = queryset.filter(date__gte=start_date)
+        queryset = queryset.order_by(ordered_by)
+        if limit:
+            queryset = queryset[:limit]
+        return queryset
+
+    @staticmethod
+    def _render_errors(errors):
+        messages = []
+        for field, field_errs in errors.items():
+            #  messages.extend([{"%s" % (field): err["message"]} for err in field_errs])
+            #  messages.extend(["in field '" + field + "': " + err["message"] for err in field_errs])
+            messages.extend([err["message"] for err in field_errs])
+        return json.dumps({"errors": messages})
+
+
+class CommaSeparatedListField(forms.CharField):
+    """ A variant of the character field that outputs cleaned data in the form of a list of strings, or None, based on
+    input delimited by commas """
+    def clean(self, value):
+        value = super().clean(value)
+        value = value.split(",")
+        if value == [""]:
+            return None
+        return value
+
+
+class MultiErrorMultipleChoiceField(forms.MultipleChoiceField):
+    """ A variant of the multiple choice field that collects errors thrown by each choice, and returns an error that
+    contains each other error. Raises up to one error per selection. Useful for validation that informs users of all
+    errors in a form at once, rather than one per form submission """
+    def validate(self, value):
+        errs = []
+        for val in value:
+            try:
+                super().validate([val])
+            except ValidationError as e:
+                errs.append(e)
+        if errs:
+            raise ValidationError(errs, code='invalid_choice')
+
+
+class ApiQueryForm(forms.Form):
+    #  overrides the constructor to interpret a string for fields as a comma separated list of choices
+    def __init__(self, *args, **kwargs):
+        if "data" in kwargs and "fields" in kwargs["data"] and isinstance(kwargs["data"]["fields"], str):
+            fields = kwargs["data"]["fields"].split(",")
+            if fields == [""]:
+                fields = None
+            kwargs["data"]["fields"] = fields
+        super().__init__(*args, **kwargs)
```

Comment:
Let's convert this to a docstring.

Reply:
I'd also recommend changing `fields` to use the `CommaSeparatedListField` as well and instead write a validator, that way we can avoid having to do this data coersion in the constructor.